sql/mysqld.cc

```diff
@@ -110,6 +110,25 @@
 #include <poll.h>
 #endif
 
+#include <my_systemd.h>
+
+#ifdef HAVE_SYSTEMD
+#include <sys/un.h>
+union sockaddr_union {
+  struct sockaddr sa;
+  struct sockaddr_in in;
+  struct sockaddr_in6 in6;
+#ifdef HAVE_SYS_UN_H
+  struct sockaddr_un un;
+#endif
+  struct sockaddr_storage ss;
+};
+#define MAX_LISTEN_SOCKETS 10
+static int systemd_listen_cnt=0;
+#else /* HAVE_SYSTEMD */
+#define MAX_LISTEN_SOCKETS 3
+#endif
```

Comment:
So, previously mysqld was listening for up to 3 sockets: base port, extra port and unix socket. Why systemd needs 10?